Algorithm.Framework/Selection/EmaCrossUniverseSelectionModel.cs

```diff
@@ -0,0 +1,102 @@
+        public EmaCrossUniverseSelectionModel(
+            int fastPeriod = 100,
+            int slowPeriod = 300,
+            int universeCount = 10000,
+            UniverseSettings universeSettings = null, 
+            ISecurityInitializer securityInitializer = null)
+            : base(false, universeSettings, securityInitializer)
+        {
+            _fastPeriod = fastPeriod;
+            _slowPeriod = slowPeriod;
+            _universeCount = universeCount;
+            _averages = new ConcurrentDictionary<Symbol, SelectionData>();
+        }
+
+        /// <summary>
+        /// Defines the coarse fundamental selection function.
+        /// </summary>
+        /// <param name="algorithm">The algorithm instance</param>
+        /// <param name="coarse">The coarse fundamental data used to perform filtering</param>
+        /// <returns>An enumerable of symbols passing the filter</returns>
+        public override IEnumerable<Symbol> SelectCoarse(QCAlgorithmFramework algorithm, IEnumerable<CoarseFundamental> coarse)
+        {
+            return (from cf in coarse
+                        // grab th SelectionData instance for this symbol
+                        let avg = _averages.GetOrAdd(cf.Symbol, sym => new SelectionData(_fastPeriod, _slowPeriod))
+                        // Update returns true when the indicators are ready, so don't accept until they are
+                        where avg.Update(cf.EndTime, cf.Price)
+                        // only pick symbols who have their _fastPeriod-day ema over their _slowPeriod-day ema
+                        where avg.Fast > avg.Slow * (1 + _tolerance)
+                        // prefer symbols with a larger delta by percentage between the two averages
+                        orderby avg.ScaledDelta descending
+                        // we only need to return the symbol and return 'Count' symbols
+                        select cf.Symbol).Take(_universeCount);
+        }
+
+        // class used to improve readability of the coarse selection function
+        private class SelectionData
+        {
+            public readonly ExponentialMovingAverage Fast;
+            public readonly ExponentialMovingAverage Slow;
+
+            public SelectionData(int fastPeriod, int slowPeriod)
+            {
+                Fast = new ExponentialMovingAverage(fastPeriod);
+                Slow = new ExponentialMovingAverage(slowPeriod);
+            }
+
+            // computes an object score of how much large the fast is than the slow
+            public decimal ScaledDelta => (Fast - Slow) / ((Fast + Slow) / 2m);
+
+            // updates the EMAFast and EMASlow indicators, returning true when they're both ready
+            public bool Update(DateTime time, decimal value) => Fast.Update(time, value) && Slow.Update(time, value);
```

Comment:
looks like some copy pasta in the comments -- I think the source algorithm this was copied from suffers from the same inconsistent comment